src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py

```diff
@@ -477,19 +485,21 @@ def convert_to_annotation(self, predictions: tuple, metadata: Dict[str, Any]) ->
         annotations = []
         height, width, _ = metadata["original_shape"]
         shape: Union[Polygon, Ellipse]
-        for score, class_idx, box, mask in zip(*predictions):
-            if score < self.confidence_threshold:
+        for obj in predictions.segmentedObjects:
+            if obj.score < self.confidence_threshold:
                 continue
             if self.use_ellipse_shapes:
-                shape = convert_bbox_to_ellipse(box[0] / width, box[1] / height, box[2] / width, box[3] / height)
+                shape = convert_bbox_to_ellipse(
+                    obj.xmin / width, obj.ymin / height, obj.xmax / width, obj.ymax / height
+                )
                 annotations.append(
                     Annotation(
                         shape,
-                        labels=[ScoredLabel(self.labels[int(class_idx) - 1], float(score))],
+                        labels=[ScoredLabel(self.labels[int(obj.id) - 1], float(obj.score))],
```

Comment:
Does this `id` mean `class_id`? Might be misleading.

Reply:
Yes, in ModelAPI result objects id = class_id

Reply:
Hope it could be fixed someday in ModelAPI repo... :)